hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/persistence/ContainerHealthSchemaManager.java

```diff
@@ -67,32 +72,43 @@ public ContainerHealthSchemaManager(
    * matching the given state will be returned.
    * @param state Return only containers in this state, or all containers if
    *              null
-   * @param offset The starting record to return in the result set. The first
-   *               record is at zero.
+   * @param minContainerId minimum containerId for filter
+   * @param maxContainerId maximum containerId for filter
    * @param limit The total records to return
    * @return List of unhealthy containers.
    */
   public List<UnhealthyContainers> getUnhealthyContainers(
-      UnHealthyContainerStates state, int offset, int limit) {
+      UnHealthyContainerStates state, Long minContainerId, Optional<Long> maxContainerId, int limit) {
```

Comment:
If limit is set but the database has fewer results than expected, ensure that the UI or client application understands when it has reached the last page (i.e., no more results). Can we add an extra field in response something like below:
`response.setHasMore(unhealthyMeta.size() == limit);`